### ISSUE TITLE
Credentials are not properly filtered out in 'bundle install'

### ISSUE THREAD

**Issue by perlun** [rubygems/rubygems#3234]
rubygems/bundler#4704 fixed some of this, but apparently there still seems to be an issue under certain circumstances.

Here is my Gemfile, running with bundler `1.16.1`:

```ruby
git_source(:acme) { |repo_name| "https://#{ENV['GITHUB_TOKEN']}:x-oauth-basic@github.com/acme/#{repo_name}" }

gem 'foobar', acme: 'foobar'
```

When I run `GITHUB_TOKEN=my-secret-token bundle install`, this is part of the output I get:

```
Using foobar 0.1.0 from https://my-secret-token:x-oauth-basic@github.com/acme/foobar (at master@abc1234)
```

This is clearly less than optimal, especially if being used on Travis or other CI services. (In our case, we are using private repos so the criticality is "less severe" _for us_, but it's still an issue since potential credential leakage is never good.)

(Rationale for using https git sources rather than SSH - this is being used in Dockerfiles, where passing in an ENV variable for a GitHub OAuth token is somewhat more convenient than having to pass in a custom SSH key.)

**Comment by perlun** [comment]
(Technically, this relates to a slightly different problem than the one fixed in #7560 if I get it right @deivid-rodriguez - it wasn't just that the private token was written to the lockfile, but it was also printed to stdout. Has this also been fixed by now?)

**Comment by deivid-rodriguez** [comment]
Nice catch @perlun. Indeed my focus in #7560 was not to fix credentials being leaked to log messages, but to the lockfile. However, I suspect #7560 has fixed this problem too, because by the time the log message is printed, the lockfile source (without credentials) has already replaced the corresponding Gemfile source (with credentials). However, worth giving this a try and reopening if not really fixed.

**Comment by perlun** [comment]
Sounds good @deivid-rodriguez. I am only a very marginal Ruby user these days, so if anyone else who is following this thread (@olleolleolle, do you copy?) has a chance to try, give it a go to see how it behaves now with the latest fixes in #7560.
